message PBMsgHeroHouseList {
  required int32 heroHouseSlots = 1;
  required int32 opennedSlot = 2;
  repeated int32 unlockSlotDesc = 3;
  
  message PBMsgSkillActived {
      required int32 id = 1;
      required bool actived = 2;
      optional int32 lv = 3;
      optional int32 hero = 4;
      optional int32 gear = 5;
      required int32 effectId = 6;
      repeated string effect = 7;
  }

  enum PBMsgHeroStatus {
      UNLOCKED = 1;
      IN_CITY_DO_NOTHING = 2;
      IN_CITY_IN_MARCH = 3;
      IN_CITY_SLEEP = 4;
      IN_NO_CITY = 5;
  }

  message PBMsgHero {
      required int64 userHeroId = 1;
      required PBMsgHeroStatus status = 2;
      required int32 heroId = 3;
      required int32 level = 4;
      required int32 renown = 5;
      required int32 nextRenown = 6;
      required int32 attack = 7;
      required int32 hp = 8;
      required int32 load = 9;
      repeated PBMsgSkillActived skill = 10;
      repeated PBMsgSkillActived fate = 11;
      required int32 might = 12;
      required int32 maxExplore = 13;
      required int32 explored = 14;
      required int32 sleepTime = 15;
      required int32 cityIndex = 16;
      required int32 slot = 17;
  }

  repeated PBMsgHero hero = 4;
}
